src/Appwrite/Platform/Modules/Compute/Base.php

```diff
@@ -335,4 +338,33 @@ public function redeployVcsSite(Request $request, Document $site, Document $proj
 
         return $deployment;
     }
+
+    /**
+     * Update manual rule for new site deployment.
+     * In case of fresh site, deployment ID will be empty in the rules, so we need to update it here.
+     *
+     * @param \Utopia\Database\Document $project
+     * @param \Utopia\Database\Document $site
+     * @param \Utopia\Database\Document $deployment
+     * @param \Utopia\Database\Database $dbForPlatform
+     * @return void
+     */
+    public function updateManualRuleForNewSiteDeployment(Document $project, Document $site, Document $deployment, Database $dbForPlatform)
+    {
+        $queries = [
```

Comment:
I would also add query for `deploymentId` being empty.

So in scenario when there is production dpeloyment, and developer makes new one, this behaviour doesnt kick in. If it did, we might be introducing downtime to deployments